presto-record-decoder/src/main/java/io/prestosql/decoder/avro/AvroColumnDecoder.java

```diff
@@ -81,14 +85,13 @@ private boolean isSupportedType(Type type)
         if (type instanceof ArrayType) {
             checkArgument(type.getTypeParameters().size() == 1, "expecting exactly one type parameter for array");
             return true;
-            //return isSupportedPrimitive(type.getTypeParameters().get(0));
         }
 
         if (type instanceof MapType) {
             List<Type> typeParameters = type.getTypeParameters();
             checkArgument(typeParameters.size() == 2, "expecting exactly two type parameters for map");
             checkArgument(typeParameters.get(0) instanceof VarcharType, "Unsupported column type '%s' for map key", typeParameters.get(0));
-            return isSupportedPrimitive(type.getTypeParameters().get(1));
+            return true;
```

Comment:
Shouldn't you recurse to `isSupportedType` here?

Also, separate this change out from `Add support for more primitive types` commit and add tests.